mvrx/src/main/kotlin/com/airbnb/mvrx/MavericksViewModelConfigFactory.kt

```diff
@@ -0,0 +1,89 @@
+package com.airbnb.mvrx
+
+import android.content.Context
+import android.content.pm.ApplicationInfo
+import androidx.annotation.RestrictTo
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlin.coroutines.CoroutineContext
+
+/**
+ * Factory for providing the [MavericksViewModelConfig] for each new ViewModel that is created.
+ *
+ * An instance of this must be set on [MvRx.viewModelConfigFactory].
+ *
+ * A custom subclass of this may be used to allow you to override [buildConfig], but this should
+ * generally not be necessary.
+ */
+open class MavericksViewModelConfigFactory(
+    /**
+     * True if debug checks should be run. Should be false for production builds.
+     * When true, certain validations are applied to the ViewModel. These can be slow and should
+     * not be used in production! However, they do help to catch common issues so it is highly
+     * recommended that you enable debug when applicable.
+     */
+    val debugMode: Boolean,
+    /**
+     * Provide a default context for viewModelScope. It will be added after [SupervisorJob]
+     * and [Dispatchers.Main.immediate].
+     */
+    val contextOverride: CoroutineContext? = null
```

Comment:
I'm still concerned that anybody who executes a coroutine within the ViewModel could still just provide their own context and override anything set here

Reply:
@gpeal yes, we can discuss separately. on use case might be to change `Dispatchers.Main.immediate` to whatever you want your default dispatcher to be. i'll likely use it for that